src/tools/auth0/handlers/emailProvider.ts

```diff
@@ -45,20 +48,18 @@ export default class EmailProviderHandler extends DefaultHandler {
     if (existing.name) {
       if (existing.name !== emailProvider.name) {
         // Delete the current provider as it's different
-        await this.client.emailProvider.delete();
+        // await this.client.emailProvider.delete();
         this.didDelete(existing);
         existing = {};
       }
     }
 
     if (existing.name) {
-      const provider = { name: emailProvider.name, enabled: emailProvider.enabled };
-      const updated = await this.client.emailProvider.update(provider, emailProvider);
+      const updated = await this.client.emails.update(emailProvider);
       this.updated += 1;
       this.didUpdate(updated);
     } else {
-      const provider = { name: emailProvider.name, enabled: emailProvider.enabled };
-      const created = await this.client.emailProvider.configure(provider, emailProvider);
+      const created = await this.client.emails.configure(emailProvider as PostProviderRequest);
```

Comment:
Can this type be applied to the construction of the `emailProvider` value above rather than smoothing-over as a type cast?

Reply:
What I mean by this is that the typing can probably be improved. I don't immediately see how but I'm sure it can to avoid this messy type casting.